src/main/java/io/fusionauth/http/server/io/HTTPInputStream.java

```diff
@@ -124,16 +158,19 @@ public int read(byte[] buffer, int offset, int length) throws IOException {
     }
 
     int read;
+    // If bodyBytes exist, they are left over bytes from parsing the preamble.
+    // - Process these bytes first before reading from the delegate InputStream.
     if (bodyBytes != null) {
-      read = Math.min(bodyBytes.length, length);
+      int remaining = bodyBytes.length - bodyBytesIndex;
+      read = Math.min(remaining, length);
```

Comment:
Fixed bug. 

The left arg to `.min` needs to take into account the current value of `bodyBytesIndex`.